src/sysen3/papsav.2

```diff
@@ -0,0 +1,385 @@
+	TYO CH
+	JUMPN WRD, 6LOOP
+	RETURN
+	.END
+
+TYPASZ:  PROCEDURE PT
+
+	REGISTER A
+
+	MOVE A,PT		;CAN'T INDIRECT PARAM
+	CALL TYPE,@A
+	RETURN
+	.END
+
+;;; PURIFICATION RITUAL
+
+pure:	0
+
+OITSVR:	0			;ITS version purified for
+OUSRST:	0			;extra check
+
+RITUAL: PROCEDURE
+
+	skipe pure
+	  jrst .+2
+	pushj p,purify
+	MOVE TT, [SQUOZE 0,ITSVRS]	;SEE IF SYSTEM CHANGED
+	.EVAL TT,
+	  JFCL
+	MOVE T, [SQUOZE 0,USRSTG]	;see if user page map changed
+	.EVAL T,
+	  JFCL
+	CAMN TT,OITSVR		;is ITS version the same>
+	  CAME T,OUSRST		;and user page map
+	    JRST NOGOOD		; nope, error out
+	RETURN			; all good, return
+
+NOGOOD:	CALL TYPE, bletch	; wrong system version, purify
+	JRST SYSMSG"KILL
+.END
+
+CONSTANTS
+VARIABLES
+
+abstb1:	<squoze 0,SYSMBF>	; system message buffer
+	sysmsg+7				
+	<squoze 0,TOIP>		; tty output ptr
+	sysmsg"syget0			
+	<squoze 0,TOBEP>	; end of buffer
+	sysmsg"syget0+2
+	<squoze 0,TOOP>		; output buffer output pointer
+	sysmsg"papsav+1
+
+immeds:	<squoze 0,TOBL>		; tty output buffer length
+	sysmsg"syget0+3
+	<squoze 0,SYSCON>	; system tty number
+	sysmsg"papsav
+	<squoze 0,SYSMLN>	; log 2 of number of 4-word blocks
+	sysmsg+6
+abstb2:
+
+PURIFY:  PROCEDURE 
+
+	REGISTER I,L,V,LIM
+
+	MOVEI LIM, IMMEDS-ABSTB1	;LIMIT ON REMAPPABLENESS
+	MOVSI I, <ABSTB1-ABSTB2>/2	;SCAN ABSTAB
+EVLOOP:	MOVE V,ABSTB1(I)		; get symbol
+	.EVAL V,			; evaluate it
+	  .VALUE			; not there???
+	CAIG LIM,(I)			; skip following code
+	  JRST EVLP00			;  if immediate symbol
+	CAIGE V,REMAPT			; remap low core
+	  SUBI V,REMAPT			; into high core
+EVLP00:	ADDI I, 1
+	MOVE TT,ABSTB1(I)		; follow patch list
+PLOOP:	skipn l,tt
+	  jrst evlpnx
+	HRRZ TT,(L)			; loc to patch
+	HRRM V,(L)			; patch it
+	JRST PLOOP			; and try again
+
+EVLPNX:	AOBJN I,EVLOOP			; next symbol
+
+; HAVING PATCHED, SET UP PAGE TABLE
+; FOR NOW, WE DON'T ACTUALLY PURIFY ANY PAGES...
+
+	setom pure
+	MOVE v, [SQUOZE 0,ITSVRS]	; get its version symbol
+	.eval v,			; evaluate it
+	  .value			; what???
+	movem v,oitsvr			; store its version
+	MOVE V, [SQUOZE 0,USRSTG]	;see if user page map changed
+	.EVAL V,			
+	  .VALUE
+	MOVEM V,OUSRST
+
+	SETZM V
+;	movei tt,1
+	MOVEI TT, REMAPT_-10.		;GET ABS PAGES
+;	hrli tt,-177
+	HRLI TT, <REMAPT-400000>_-10.	;FROM 0 TO 400000
+	MOVE T, TT
+;GETMOR: .CALL CORBLK,[%climm,,%cbred+%cbndr,%climm,,%jself,TT,%climm,,400000,T]
+GETMOR:	.CALL CORBLK,[#210000,#-1,TT,#400000,T]
+	  JRST .+2		; error return
+	JRST GOTIT
+
+	.SUSET [.RBCHN,,V]	; find out what lost
+	.CALL STATUS,V,V	; get error code
+	  .VALUE
+	LDB V,[270600,,V]
+	CAIE V,%eropg		; can't get that address
+  	  .VALUE		; no - unclear
+	ADD TT,[40,,40]		; yes - biting MD10's have a 32K hole!
+	MOVE T,TT
+	JUMPL TT,GETMOR
+
+GOTIT:	skipe v
+	CALL TYPE,MEMHOL	; type memory hole message
+	MOVEI TT, <1000000-REMAPT>_-10.	;DO REMAPPAGE
+	HRLI TT, -<REMAPT_-10.>
+	SETZ T,
+;	.CALL CORBLK,[#%climm,,%cbred+%cbndr,#%climm,,%jself,TT,#%climm,,400000,T]
+	.CALL CORBLK,[#210000,#-1,TT,#400000,T]
```

Comment:
I'd suggest to remove the CORBLK comments for now, and update the calls in another iteration.

Reply:
Oh, then I understand why you left the comments.